### PR TITLE
Profile/aw/53630/ profile use accessible info card mvp

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

As the first set of work in addressing accessibility issues, this PR creates a new `ProfileInfoCard` component and uses it in the Direct Deposit page of profile, but also places it behind a feature toggle.

The component is only used in one section for now within the 'Payment History' section at the bottom of the page.

Additional work in this PR:

- Removes the `prefixUtilityClasses` function usage and uses the classNames package instead to compose class names.
- Introduces a new experimental Feature Toggle helper component called `Toggler`
  - A Toggler instance can be imported as a single component, and exposes an Enabled and Disabled compound components that can be used as children to render conditional content.
  - In addition to the components, there is `Toggler.TOGGLE_NAMES` that is a property that contains all the toggle names from the profile `featureFlagNames` so that just a single import can be used without additional redux mapStateToProps or selectors are needed in any way.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/53630

## Testing done

- Added unit tests for Toggler component
- Visually tested ProfileInfoCard to verify it displays the same as the ProfileInfoTable

- Will be adding further tests for more complex lists as they are worked on. This is MVP at this point and already pushing the 500 LOC PR recommended max diff.

## Screenshots
BEFORE

Screenshot 2023-02-20 at 1 22 18 PM

AFTER

Screenshot 2023-02-20 at 1 17 33 PM

## What areas of the site does it impact?
Profile application - Direct Deposit page, and just the last section that displays 'VA Payment History' section.

## Acceptance criteria

- [x] Component built to display visually identical to ProfileInfoTable
- [x] Component guarded behind feature toggle
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  I added a screenshot of the developed feature
- [x]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

Curious to hear people's thoughts on the Toggler :) I'm sure it wouldn't cover complex use cases for toggles on the FE, but was also thinking about implementing an additional hook that could be imported to supply a more streamlined primitive for advanced usages.

### CODE REVIEW

File: src/applications/personalization/profile/mocks/feature-toggles/index.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const generateFeatureToggles = (toggles = {}) => {
     profileShowPaymentsNotificationSetting = false,
     profileBlockForFiduciaryDeceasedOrIncompetent = false,
     profileUseSecurityProcessList = true,
+    profileUseInfoCard = false,
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/mocks/feature-toggles/index.js:17]: To test this work, you can set the profileUseInfoCard to true and load the Direct Deposit page.

File: src/applications/personalization/profile/sass/profile-info-table.scss [hunk 2]
```diff
@@ -1,6 +1,6 @@
 $corner-size: 4px;
 
-.profile-info-table {
+.profile-info-table, .profile-info-card {
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/sass/profile-info-table.scss:3]: For this scss file in general, if we can determine that the border radius isn't needed on this table, then we _could_ get rid of this whole file (the border is the only custom part of this really). 

I'm confident that the VA design system classes can be composed to accomplish the same goals without custom styling.

File: src/applications/personalization/profile/components/ProfileInfoCard.jsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { numberBetween } from '../../common/proptypeValidators';
+
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/ProfileInfoCard.jsx:6]: For the classNames below. 

This is deprecating the previously used `prefixUtilityClasses` function. I found the old utility usage somewhat difficult to read, and just not ideal in my mind. The new composition of classNames uses one of the already well established packages [classNames](https://github.com/JedWatson/classnames) and assembles them with their full names, and I thought that was easier to read and manipulate.

I didn't remove the function from the platform code (although its only really used in one other place), but will evaluate it further and decide if that is something I want to do within the profile code after this launches.

File: src/applications/personalization/profile/components/ProfileInfoCard.jsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+const sectionTitle = classNames([
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/ProfileInfoCard.jsx:7]: For the classNames below. 

This is deprecating the previously used `prefixUtilityClasses` function. I found the old utility usage somewhat difficult to read, and just not ideal in my mind. The new composition of classNames uses one of the already well established packages [classNames](https://github.com/JedWatson/classnames) and assembles them with their full names, and I thought that was easier to read and manipulate.

I didn't remove the function from the platform code (although its only really used in one other place), but will evaluate it further and decide if that is something I want to do within the profile code after this launches.

File: src/applications/personalization/profile/components/ProfileInfoCard.jsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  children: PropTypes.node,
+  className: PropTypes.string,
+  level: numberBetween(1, 6),
+  namedAnchor: PropTypes.string,
+};
+
+export const List = ({ data }) => {
+  return (
+    <>
+      {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/personalization/profile/components/ProfileInfoCard.jsx:93]: this was from previous implementation as accessibility concern i guess 🤷🏼

**Comment by adamwhitlock1** [comment]
@allisonlu I actually went ahead and added a common hooks folder and a useFeatureToggle hook, that can be used now. Will follow up with some tests for that hook as well.

**Review by allisonlu — APPROVED** [review]
lgtm! looking forward to using the Toggler component and the new hook

**Comment by adamwhitlock1** [comment]
just an fyi on this PR, I did add unit tests for the useFeatureToggle hook, but no core functionality was changed in the last commit.

File: src/applications/personalization/profile/tests/components/Toggler.unit.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/personalization/profile/tests/components/Toggler.unit.spec.jsx:1]: this is for camel cased feature toggle names

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**